flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,250 @@
+            logger.verbose('TLS version is {}', tls_version)
+
+            cert_required = (
+                ssl.CERT_REQUIRED if certs.get('validate_broker_cert', True) else ssl.CERT_NONE
+            )
+
+            client.tls_set(
+                ca_certs=certs.get('broker_ca_cert'),
+                certfile=certs.get('client_cert'),
+                keyfile=certs.get('client_key'),
+                cert_reqs=cert_required,
+                tls_version=tls_version,
+            )
+
+            if not certs.get('validate_broker_cert'):
+                client.tls_insecure_set(True)
+                logger.debug('TLS insecure cert mode enabled. Broker cert will not be validated')
+            else:
+                logger.debug('TLS secure cert mode enabled. Broker cert will be validated')
+
+        # Handle user/pass authentication
+        if config.get('username'):
+            logger.debug('Credential passwords s are redacted to protect the innocent...')
+            logger.debug(
+                'Auth credentials: username=[{}] password sha256 hash is "{}"',
+                config.get('username'),
+                sha256(str(config.get('password')).encode('utf-8')).hexdigest(),
+            )
+            logger.debug(
+                'You can validate them yourself by calculating the sha256 hex digest of your password string (google is your friend if you do not know how to do this)'
+            )
+            logger.debug(
+                'Note: a password that is not provided (i.e. None) will hash to "{}"',
+                sha256(str(None).encode('utf-8')).hexdigest(),
+            )
+
+            client.username_pw_set = (config.get('username'), config.get('password'))
+
+        try:
+            logger.verbose(
+                "Connecting to {}:{}", config.get('broker_address'), config.get('broker_port')
+            )
+            client.connect(
+                config.get('broker_address'),
+                config.get('broker_port'),
+                config.get('broker_timeout'),
+            )
+            logger.verbose("Connected to MQTT broker")
+        except Exception as e:
+            raise PluginWarning(f'Error connecting to MQTT broker: {e}')
+
+        try:
+            logger.verbose(
+                'Publishing message [{}] to topic [{}] ',
+                config.get('payload'),
+                config.get('topic'),
+            )
+            publish_info = client.publish(
+                config.get('topic'),
+                config.get('payload'),
+                qos=config.get('qos'),
+                retain=config.get('retain'),
+            )
+            logger.verbose(
+                "Notification sent to broker, waiting for callback response to confirm publishing success - rc={}",
+                publish_info,
+            )
+        except Exception as e:
+            raise PluginWarning(f'Error publishing to MQTT broker: {e}')
```

Comment:
Should probably use log.exception .. this catches everything unknown and with this we don't get proper tracebacks.